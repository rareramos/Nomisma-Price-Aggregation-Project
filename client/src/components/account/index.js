export { Account as default} from './account-container';
