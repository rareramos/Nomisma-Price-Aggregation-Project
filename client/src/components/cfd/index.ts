export { Cfd } from './cfd-container';
