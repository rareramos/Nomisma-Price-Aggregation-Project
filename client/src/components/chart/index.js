export { Chart } from './chart-container';
