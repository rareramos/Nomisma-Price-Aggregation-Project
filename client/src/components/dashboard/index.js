export { Dashboard } from './dashboard-container';
