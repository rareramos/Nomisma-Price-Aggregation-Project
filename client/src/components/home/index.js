export { Home as default } from './home-container';
