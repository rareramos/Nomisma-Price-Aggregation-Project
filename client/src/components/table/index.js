export { CustomPaginationTable } from './table-container';
