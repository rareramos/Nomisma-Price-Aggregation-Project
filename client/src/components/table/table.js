import React, { Component } from 'react';
import { object } from 'prop-types';
import { BootstrapTable, TableHeaderColumn } from 'react-bootstrap-table';

export class CustomPaginationTable extends Component {
  renderShowsTotal(start, to, total) {
    return (
      <p style={{ color: 'blue' }}>
        From { start } to { to }, totals is { total }&nbsp;&nbsp;(its a customize text)
      </p>
    );
  }

  render() {
    const rates = this.props.tableData ? this.props.tableData.rates.dataSlice : '';

    const options = {
      page: 3,  // which page you want to show as default
      sizePerPageList: [ {
        text: '5', value: 5,
      }, {
        text: '10', value: 10,
      }, {
        text: 'All', value: rates.length,
      } ], // you can change the dropdown list for size per page
      sizePerPage: 5,  // which size per page you want to locate as default
      pageStartIndex: 0, // where to start counting the pages
      paginationSize: 3,  // the pagination bar size.
      prePage: 'Prev', // Previous page button text
      nextPage: 'Next', // Next page button text
      firstPage: 'First', // First page button text
      lastPage: 'Last', // Last page button text
      paginationShowsTotal: this.renderShowsTotal,  // Accept bool or function
      paginationPosition: 'top',  // default is bottom, top and both is all available
      // hideSizePerPage: true > You can hide the dropdown for sizePerPage
      // alwaysShowAllBtns: true // Always show next and previous button
      // withFirstAndLast: false > Hide the going to First and Last page button
    };
    return (
      <BootstrapTable
        data={ rates }
        pagination
        options={ options }
      >
        <TableHeaderColumn
          dataField='creationTime'
          isKey
        >Product ID</TableHeaderColumn>
        <TableHeaderColumn dataField='id'>Product Name</TableHeaderColumn>
        <TableHeaderColumn dataField='loanTerm'>Product Price</TableHeaderColumn>
      </BootstrapTable>
    );
  }
}

CustomPaginationTable.propTypes = {
  tableData: object,
};
