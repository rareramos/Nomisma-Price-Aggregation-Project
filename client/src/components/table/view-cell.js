import React from 'react';
import { string } from 'prop-types';

export const ViewCell = ({
  transactionHash,
}) => (
  <a
    href={`https://etherscan.io/tx/${transactionHash}`}
    target="_blank"
    rel="noopener noreferrer"
  >
    View
  </a>
);

ViewCell.propTypes = {
  transactionHash: string.isRequired,
};
