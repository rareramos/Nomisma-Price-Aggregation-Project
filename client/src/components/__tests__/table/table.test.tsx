import * as React from 'react';
import { shallow } from 'enzyme';

import { ViewCell } from 'components/table/view-cell';
import { IViewCellProps } from 'types/components/table';

const ViewCellProps : IViewCellProps = {
  transactionHash: 'usd',
};

describe('Table components tests', () => {
  it('matches Viewcell snapshot', () => {
    const wrapper = shallow(<ViewCell {...ViewCellProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
