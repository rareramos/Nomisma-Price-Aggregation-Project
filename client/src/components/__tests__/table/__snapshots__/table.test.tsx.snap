// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table components tests matches Viewcell snapshot 1`] = `
<a
  href="https://etherscan.io/tx/usd"
  rel="noopener noreferrer"
  target="_blank"
>
  View
</a>
`;
